hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/discovery/DiscoveryAddressTranslator.java

```diff
@@ -61,7 +61,7 @@ public Address translate(Address address) {
 
         privateToPublic = this.privateToPublic;
         Address publicAddress = privateToPublic.get(address);
-        if (!alreadyRefreshed) {
+        if (publicAddress == null && !alreadyRefreshed) {
```

Comment:
In most of the calls alreadyRefreshed will be `false`. In that case, we don't want to refresh if public address is not null. Hence it should be &&